### ISSUE TITLE
can't start django server

### ISSUE THREAD

**Issue by kang000feng** [openshift/django-example#52]
hi openshift,
I deploy django with your quick start : https://www.openshift.com/quickstarts/django , then i try to start django server with  python manage.py runserver in SSH TERMINAL, keep on getting the following error:
Performing system checks... 
System check identified no issues (0 silenced). 
October 04, 2015 - 20:56:45 
Django version 1.8.2, using settings 'mysite.settings' 
Starting development server at http://127.0.0.1:8000/ 
Quit the server with CTRL-BREAK. 
Error: You don't have permission to access that port. 

how can i start django server successfully with  python manage.py runserver? or is there any way to start it?

**Comment by j1fig** [comment]
Hi @kang000feng ,

You don't need to run `python manage.py runserver` on Openshift, your application should already be running. In fact it's [recommended that you don't do so](https://docs.djangoproject.com/en/1.8/howto/deployment/wsgi/modwsgi/).

This quickstart runs on python cartridge, which is:

* running an Apache with mod_wsgi server
* forwarding requests to your application WSGI entrypoint, i.e. your `application` defined in your wsgi.py

This Apache web server binds to ports 80 (and 443 for HTTPS, though here SSL is terminated at web server) and forwards requests sent to these ports to your application (except maybe for file requests which are served directly by Apache here I think).

I think that in your execution context you're only allowed to have an application binding to port 80, which is already being used by Apache. Hence the error you're seeing.

So, in short: when your gear starts, it starts a process which is looking for your app's wsgi.py file and running your web app. Just running this donesn't work for you? Can you paste the `rhc tail` result?

**Comment by kang000feng** [comment]
thank you João! the point is i am installing a  python Agent in my django of the openshift, and it requires embeding agent and rebooting application with command line tool, can you pls help me out? 
 is there any way to start my application from rhc or ssh window? thanks very much!

**Comment by j1fig** [comment]
Hi @kang000feng ,

Not sure what you're trying to accomplish, but to restart an application from `rhc` you can do

    rhc app stop <your_app_name> -n <your_domain_name>
    rhc app start <your_app_name> -n <your_domain_name>

There is also `rhc restart` but I've found that it doesn't reload the environment variables and Python environment. So it's best to do `stop` and then `start` to be sure.

Note: This will make your site temporarily unavailable of course ;)

Good luck,
Joao

**Comment by kang000feng** [comment]
thank you Joao, i really appreciate it. what command i can use to restart my application from putty?

**Comment by j1fig** [comment]
If by putty you mean  an SSH session into the gear I'd say:

    ctl_app restart

and then select the cartridge to restart.

**Comment by kang000feng** [comment]
yes, thank you Joao so much! i will try it later. so there is no way to run 'python manage.py runserver' to restart my application?

**Comment by kang000feng** [comment]
hi Joao ，may i ask how my app bind port 8080 to $OPENSHIFT_INTERNAL_IP? which file i need to modify? what is the code excatly ？

**Comment by j1fig** [comment]
Hi @kang000feng . That I don't know. I guess you could try #openshift of freenode IRC, or opening a ticket if you have Silver plan.
